packages/ui-extensions/src/surfaces/checkout/preact/subscription.ts

```diff
@@ -0,0 +1,45 @@
+import {useEffect, useState} from 'preact/hooks';
+import type {StatefulRemoteSubscribable} from '@remote-ui/async-subscription';
```

Comment:
Could we just copy this instead of pulling it in? I would like to remote the remote-ui dependency if we can

Reply:
I think this is a valid point. There's only a few relevant types we would need.